File: java/monsters/options/Creature.java
package monsters.options;

public class Creature implements Fightable,Damagable{
    protected int power;
    protected int penalty;

    public Creature(int power, int penalty) {
        this.power = power;
        this.penalty = penalty;
    }

    @Override
    public void takeDamage() {
        if(power - penalty < 0) power = 0;
        else power -=penalty;
    }

    @Override
    public int attack() {
        return power;
    }
}

File: php/monsters/options/Creature.php
<?php

class Creature implements Fightable, Damagable {
    protected $power;
    protected $penalty;

    public function __construct($power, $penalty) {
        $this->power = $power;
        $this->penalty = $penalty;
    }

    public function takeDamage() {
        if ($this->power - $this->penalty < 0) {
            $this->power = 0;
        } else {
            $this->power -= $this->penalty;
        }
    }

    public function attack() {
        return $this->power;
    }
}

?>